### DISCUSSION TITLE
What is the difference between "liveReload" and "watch" in webpack devServer configuration?

### DISCUSSION THREAD

**Discussion by SetTrend** [webpack/webpack#15299] [Q&A] [answered]
What is the difference between [`devServer.liveReload`](https://webpack.js.org/configuration/dev-server/#devserverlivereload) and [`watch`](https://webpack.js.org/configuration/dev-server/#watch) configuration in particular?

Both seem to serve the same purpose, and – according to documentation – none of them does anything without the other one being enabled? The documentation appears to be a bit blurred on this.

**Comment by alexander-akait** [comment, marked as answer]
Yes, if you want to reload browser on changes in external files (not in compilation) you need to specify both, but there are cases when you can disable `liveReload` (you can have own logic for reloading, it is for low level configuration with custom solutions), by default `liveReload` is `true`, so you don't worry, just setup `watchFiles` if you need it
